### PR TITLE
NODE-1080: Create MongoNetworkError

### PR DESCRIPTION
**Pull request by sebthedev**
This PR creates the new MongoNetworkError class of errors. Having MongoNetworkError allows us to identify which errors are resumable errors for the purposes of Change Streams (see [NODE-1057](https://jira.mongodb.org/browse/NODE-1057)).

### CODE REVIEW

File: lib/network_error.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+"use strict";
```
Comment (mbroadst) [inline lib/network_error.js]: Is there a reason to have this live in its own file? Afaict, it should just live inside `error` and probably actually subclass `MongoError` itself (this is a specialization of that type)

Comment (sebthedev) [inline lib/network_error.js]: I was unsure how to best do this. I'll research subclassing.

File: lib/error.js [hunk 2]
```diff
@@ -1,44 +1,65 @@
...
+  } else {
+    if (typeof message === 'string') {
+      this.message = message;
+    } else {
+      this.message = message.message || message.errmsg || message.$err || 'n/a';
+      for (var name in message) {
+        this[name] = message[name];
+      }
+    }
+    Error.captureStackTrace(this, MongoError);
```
Comment (mbroadst) [inline lib/error.js]: also I think we _always_ want to capture the stack trace. If an error is passed in, you should include that as like an originating error, it shouldn't _become_ this error

Comment (mbroadst) [inline lib/error.js]: actually forget that, I guess that is the original intent...

**Pull request merged**